crawl-ref/source/abyss.cc

```diff
@@ -1808,14 +1831,30 @@ static bool _incorruptible(monster_type mt)
     return mons_is_abyssal_only(mt) || mons_class_holiness(mt) == MH_HOLY;
 }
 
+static bool _incorruptible_weaker(monster_type mt)
+{
+    return mons_is_abyssal_only(mt) || mons_class_holiness(mt) == MH_HOLY
+    || mons_demon_tier(mt) == 1     || mons_demon_tier(mt) == 2
+    || mt == MONS_TENTACLED_STARSPAWN;
+    // no tier 1 or 2 demons, or starspawn for the monster version
+}
+
 // Create a corruption spawn at the given position. Returns false if further
 // monsters should not be placed near this spot (overcrowding), true if
 // more monsters can fit in.
-static bool _spawn_corrupted_servant_near(const coord_def &pos)
+static bool _spawn_corrupted_servant_near(const coord_def &pos, bool is_monster_version)
```

Comment:
Weaving this `is_monster_version` flag through everywhere makes me nervous. I'd really consider splitting the function entirely, especially if you take my suggestion of making the monster version have fewer effects on terrain (or even only cosmetic effects).

Reply:
Yes agreed - it's a little bit more code duplication that way but easier to follow/update in future and does fit better with the two effects being distinct in nature. I've split this out now.